core/src/main/java/io/grpc/Deadline.java

```diff
@@ -141,4 +141,17 @@ public long timeRemaining(TimeUnit unit) {
   public String toString() {
     return timeRemaining(TimeUnit.NANOSECONDS) + " ns from now";
   }
+
+  @Override
+  public int compareTo(Deadline that) {
```

Comment:
We should probably have a few tests for this. They should be pretty easy to make; no need to go all out.